src/lib.rs

```diff
@@ -1696,6 +1696,86 @@ pub fn strongly_connected_components(
         .collect()
 }
 
+/// Return the first cycle encountered during DFS of a given PyDiGraph
+///
+/// :param PyDiGraph graph: The graph to find the cycle in
+/// :param int root: Optional index for starting node for cycle
+///
+/// :returns: A list describing the cycle. The index of node ids which
+///     forms a cycle (loop) in the input graph
+/// :rtype: list
+#[pyfunction]
+#[text_signature = "(graph, /, root=None)"]
+pub fn digraph_find_cycle(
+    graph: &digraph::PyDiGraph,
+    root: Option<usize>,
+) -> Vec<(usize, usize)> {
+    let mut root_node = root;
+    let mut graph_nodes: HashSet<NodeIndex> =
+        graph.graph.node_indices().collect();
+    let mut cycle: Vec<(usize, usize)> = Vec::new();
+    let temp_value: NodeIndex;
+    // If root_node is not set get an arbitrary node from the set of graph
+    // nodes we've not "examined"
+    let root_index = match root_node {
+        Some(root_value) => NodeIndex::new(root_value),
+        None => {
+            temp_value = *graph_nodes.iter().next().unwrap();
+            graph_nodes.remove(&temp_value);
+            temp_value
+        }
+    };
+
+    // Stack (ie "pushdown list") of vertices already in the spanning tree
```

Comment:
Heh, this comment was kind of a joke and mostly made made sense in the context of `cycle_basis` functions. The paper describing the algorithm for `cycle_basis` is quite old and called a stack a pushdown list, which is a descriptive name but not one commonly used because it was before CS terminology was standardized. But I'm glad to see it catching on :)

Reply:
Hah! Let's keep it then